File: entites/Thing.py
class Thing:
    def __init__(self, x, y, angle, doomType, flags, skillLevel12, skillLevel3, skillLevel45, deaf, notSinglePlayer):
        self.x = x
        self.y = y
        self.angle = angle
        self.doomType = doomType
        self.flags = flags
        self.skillLevel12 = skillLevel12
        self.skillLevel3 = skillLevel3
        self.skillLevel45 = skillLevel45
        self.deaf = deaf
        self.notSinglePlayer = notSinglePlayer

    def __str__(self):
        return f"{self.x=}, {self.y=}, {self.angle=}, {self.doomType=}, {self.flags=}, {self.skillLevel12=}, {self.skillLevel3=}, {self.skillLevel45=}, {self.deaf=}, {self.notSinglePlayer=}"

    __repr__ = __str__

File: cpp_code/Thing.cpp
#include "./headers/Thing.h"
#include <cstdint>
#include <iostream>


Thing::Thing() {
    Thing(0,0,0,0,0,false,false,false,false,false);
}

std::ostream& operator<< (std::ostream& os, const Thing& obj) {
        os << "Thing{ ";
        os << "x: " << obj.x << " ";
        os << "y: " << obj.y << " ";
        os << "angle: " << obj.angle << " ";
        os << "doomType: " << obj.doomType << " ";
        os << "flags: " << obj.flags << " ";
        os << "skillLevel12: " << obj.skillLevel12 << " ";
        os << "skillLevel3: " << obj.skillLevel3 << " ";
        os << "skillLevel45: " << obj.skillLevel45 << " ";
        os << "deaf: " << obj.deaf << " ";
        os << "notSinglePlayer: " << obj.notSinglePlayer << " ";
        os << "}";
        return os;
    }  